### ISSUE TITLE
Make backend configuration optional when using backend type `azurerm`

### ISSUE THREAD

**Issue by charleszipp** [jason-johnson/azure-pipelines-tasks-terraform#15]
Given I have the following terraform template

```terraform
terraform {
  backend "azurerm" {
    resource_group_name  = "rg-name"
    storage_account_name = "stwhatever"
    container_name       = "terraformstate"
    key                  = "some.terraform.tfstate"
  }
}
```

I would like to be able to execute terraform init with a service connection but, no backend configuration since its already provided in the template

```yaml
- task: TerraformCLI@0
  inputs:
    command: 'init'
    backendType: 'azurerm'
    backendServiceArm: 'my-service-connection'
```

** Note; In this case, ensureBackend may not be possible given it needs the storage sku and location to create properly. The user would need to provide the SKU and Location for ensureBackend to work.

Replaces charleszipp/azure-pipelines-tasks-terraform-old#253

**Comment by markphahn** [comment]
I also have the use case and would like to see this addressed.

**Comment by odoo-ce-modules** [comment]
If I understand this issue correctly, as the documentation says "[If azurerm selected, the task will prompt for a service connection and storage account details to use for the backend.]" ... which basically means that its currently IMPOSSIBLE to use the Terraform backend [Partial Configuration](https://www.terraform.io/language/settings/backends/configuration#partial-configuration) feature at all with this? 

With Terraform's Partial Configuration feature, we would simply specify in commandOptions, `-backend-config=PATH` .. where PATH is the path to the backend.conf file to use for the rest of the configuration. This is very unfortunate. I see this project has forks .. are you accepting help from the public?

**Comment by charleszipp** [comment]
released with 0.7.2
